### ISSUE TITLE
Details summary margin appears to be wrong

### ISSUE THREAD

**Issue by facelessuser** [squidfunk/mkdocs-material#1995]
__I checked that...__

- [x] ... the documentation does not mention anything about my problem
- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... there are no open or closed issues that are related to my problem

## Description

It appears the margin for summary is incorrect. It appears to use

```css
.md-typeset summary {
    margin: 0 -0.6rem 0 -0.8rem;
}
```

But should use:

```css
.md-typeset summary {
    margin: 0 -0.6rem 0 -0.6rem;
}
```

![detail-bug](https://user-images.githubusercontent.com/1055125/96288869-5713d180-0fa1-11eb-998f-da52580d89e0.gif)

The problem doesn't appear until you try to override the background color of the summary. The above change does not affect the visual appearance of the base style, which seems to suggest that it is a quirk due to when the style is applied.

To demonstrate that this is not related to my site's custom styling, I've demonstrated that it occurs
on Materials official docs with a simple CSS change:

<img width="771" alt="Screen Shot 2020-10-16 at 11 18 05 AM" src="https://user-images.githubusercontent.com/1055125/96289044-980be600-0fa1-11eb-88f4-60c5a8abf739.png">

I have replicated this on both Chrome and Firefox. I have not tried others yet.

### Expected behavior

Changing the summary background should not make the left details border disappear.

### Actual behavior

Summary background changes should not cause the left detail's border to disappear.

### Steps to reproduce the bug

See description

### Package versions

* Python: 3.8
* MkDocs: Latest
* Material: 6.0

### System information

* OS: macOS
* Browser: Firefox and Chrome, maybe more.

**Comment by squidfunk** [comment]
The way the summary is rendered was changed in the process of revamping accessibility, specifically the focus of elements. With the changes, the focus outline will correctly span all to the left over the border:

<img width="710" alt="Bildschirmfoto 2020-10-17 um 11 01 29" src="https://user-images.githubusercontent.com/932156/96333158-978a5280-1068-11eb-83d7-522d9f9a95d1.png">

Before, it didn't – it only spanned the background, leaving the border out of the focus – and that looked weird. However, you can easily bring back the old style by setting `border-left` on the `summary` as well, e.g.:

``` css
... {
  border-left: .2rem solid #448aff;
  background: yellow;
}
```

<img width="705" alt="Bildschirmfoto 2020-10-17 um 11 08 42" src="https://user-images.githubusercontent.com/932156/96333263-2c8d4b80-1069-11eb-8686-4e16b74d9227.png">

This will also retain the new focus outline behavior.

**Comment by squidfunk** [comment]
For reference, here's how it looked before:

<img width="567" alt="Bildschirmfoto 2020-10-17 um 16 15 02" src="https://user-images.githubusercontent.com/932156/96339273-f74a2300-1093-11eb-9df3-ec799efb3ac9.png">

**Comment by facelessuser** [comment]
Hmm, that is surprising and unintuitive from a styling override perspective. I wonder why summary only needs the left border style when you override the background. That is peculiar behavior. Though it seems to be consistent across browsers 🤷 .

> For reference, here's how it looked before:

Is that a problem?

I know we are diving into personal opinion here. I'm not pressing to have it changed if you feel the current style is sane, but I suspect you may get this question more in the future.

**Comment by squidfunk** [comment]
It needs the style because the default Admonition backgrounds are transparent versions of the same color, thus not layering over it, not changing it. 8eb23755 adds the border to the summary. What sucks about this is the fact that we could in theory just inherit the border on the summary, omitting the needed to define all Admonition color shades for each type. However, this does not work in Chrome due to a bug ([which I just found out about](https://twitter.com/squidfunk/status/1317466251996876800)). For this reason, this change will now slightly increase the CSS size.

**Comment by facelessuser** [comment]
> However, this does not work in Chrome due to a bug (which I just found out about)

Yeah, the quirks of different browsers 😬.

If nothing, this will make it more obvious what is happening. The user will have something they can see that is not getting styled.

**Comment by squidfunk** [comment]
Released as part of 6.1.0.
